File: ArchitectureComponentsMVVM/app/src/main/java/com/andr/mvvm/repository/UserDao.java
package com.andr.mvvm.repository;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;
@Dao
public interface UserDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(User user);

    @Query("SELECT * from user_info ORDER BY user_name ASC")
    LiveData<List<User>> getAllUsers();

    @Query("DELETE FROM user_info")
    void deleteAll();
}

File: MVVM_Kotlin/app/src/main/java/com/mvvm/kot/Repository/UserDao.kt
package com.mvvm.kot.Repository

import android.arch.lifecycle.LiveData
import android.arch.persistence.room.Dao
import android.arch.persistence.room.Insert
import android.arch.persistence.room.OnConflictStrategy
import android.arch.persistence.room.Query
import com.mvvm.kot.ViewModels.User

@Dao
interface UserDao {

@Insert(onConflict = OnConflictStrategy.REPLACE)
fun insert(user:User)

@Query("SELECT * from user_info ORDER BY user_name ASC")
fun getAllUsers():LiveData<List<User>>

@Query("DELETE FROM user_info")
fun deleteAll()
}